reporter/iface.go

```diff
@@ -87,10 +87,16 @@ type SymbolReporter interface {
 	// open the file and then enqueue the upload in the background.
 	ExecutableMetadata(args *ExecutableMetadataArgs)
 
+	// FrameKnown may be used to query the reporter if the FrameID is known. This is mainly
+	// used to see if the data is already cached and extra work to resolve the metadata
+	// should not be done. If the reporter returns false, the interpreter plugins will
+	// resolve the frame metadata and submit it to the reporter via a FrameMetdata call.
+	FrameKnown(frameID libpf.FrameID) bool
+
 	// FrameMetadata accepts metadata associated with a frame and caches this information before
 	// a periodic reporting to the backend.
-	FrameMetadata(fileID libpf.FileID, addressOrLine libpf.AddressOrLineno,
-		lineNumber libpf.SourceLineno, functionOffset uint32, functionName, filePath string)
+	FrameMetadata(frameID libpf.FrameID, lineNumber libpf.SourceLineno,
```

Comment:
nit: can we convert this list of arguments to a single struct containing these arguments?
This could help in the long run adding more elements to the struct without breaking the package API.

Reply:
Yes, I was considering this already when writing this PR, but was wondering if this becomes too intrusive. Since we are touching the API and its mechanical change, perhaps it's better to just do it now. I'll amend this PR.